ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/helpers/BeaconStateAccessors.java

```diff
@@ -90,6 +91,22 @@ public Optional<BLSPublicKey> getValidatorPubKey(BeaconState state, UInt64 valid
                 }));
   }
 
+  public boolean validateWithdrawalCredentials(
+      BeaconState state, BlsToExecutionChange addressChange) {
+    final int validatorIndex = addressChange.getValidatorIndex().intValue();
+    if (state.getValidators().size() <= validatorIndex || validatorIndex < 0) {
```

Comment:
A UInt64 can't be  `< 0` and `intValue` checks it hasn't overflowed so we can skip the `validatorIndex < 0` check here.

Also `validatorIndex` must be strictly less than `validators.size()` since the indices start from 0.

The spec has it right: `assert address_change.validator_index < len(state.validators)`